### PR TITLE
docs: translation for advanced/plugin.md

### PR DESCRIPTION
**Pull request by AdoBeatTheWorld**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ x] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

英文翻译 docs/source/en/advanced/plugin.md

### CODE REVIEW

File: docs/source/en/advanced/plugin.md [hunk 1]
```diff
@@ -1 +1,467 @@
-this document is still waiting for translation, see [Chinese Version](/zh-cn/advanced/plugin.html)
\ No newline at end of file
+
+title: Plugin Development
+---
+
+Plugins is the most important feature in Egg framework. It keeps Egg simple, stable and effeciently, and also it can make the best reuse of bussiness logic, to build an entire ecosystem. Some should be get confused
```
Comment (SunShinewyf) [inline docs/source/en/advanced/plugin.md]: effeciently -> efficient (这里使用形容词）
bussiness -> business
Some should be get confused -> some people may be confused 感觉这样会好一点

File: docs/source/en/advanced/plugin.md [hunk 2]
```diff
@@ -1 +1,467 @@
...
+
+- Since Koa have already got plugins, what's the point of the Egg plugins
```
Comment (SunShinewyf) [inline docs/source/en/advanced/plugin.md]: 中文是 koa 已经有中间件的机制 ,所以前半句不太符合中文意思，plugins -> middlewares , have -> has

File: docs/source/en/advanced/plugin.md [hunk 3]
```diff
@@ -1 +1,467 @@
...
+- What is the differences between middleware, plugin and application, what is there relationships
```
Comment (SunShinewyf) [inline docs/source/en/advanced/plugin.md]: is -> are （注意单复数）
后半句感觉有点不顺，建议： and what is the relationship

File: docs/source/en/advanced/plugin.md [hunk 4]
```diff
@@ -1 +1,467 @@
...
+- how do i use the plugin
+- how do i build a plugin
```
Comment (SunShinewyf) [inline docs/source/en/advanced/plugin.md]: 这里的 i 建议换成 I

File: docs/source/en/advanced/plugin.md [hunk 5]
```diff
@@ -1 +1,467 @@
...
+- ...
+
+As we've already explained some these points in Chapter [using plugins](../basics/plugin.md) before. Now we are going through how to build a plugin.
+
+## Plugin Development
+
+### Quick Start with Scaffold
+
+you can chose [plugin][egg-boilerplate-plugin] scaffold in [egg-init] for quick start.
+
```
Comment (SunShinewyf) [inline docs/source/en/advanced/plugin.md]: chose -> choose （注意动词时态）

File: docs/source/en/advanced/plugin.md [hunk 6]
```diff
@@ -1 +1,467 @@
...
+|   ├── config.test.js (optional)
+|   ├── config.local.js (optional)
+|   └── config.unittest.js (optional)
+└── test
+    └── middleware
+        └── mw.test.js
+```
+
+it is very likely to the application directory, what's the difference?  
+
```
Comment (SunShinewyf) [inline docs/source/en/advanced/plugin.md]: 这里建议改为： it is almost the same as the directory of application

File: docs/source/en/advanced/plugin.md [hunk 7]
```diff
@@ -1 +1,467 @@
...
+1. Plugin have no seperated router and controller. this is because:
```
Comment (SunShinewyf) [inline docs/source/en/advanced/plugin.md]: seperated -> separated (注意单词拼写）

File: docs/source/en/advanced/plugin.md [hunk 8]
```diff
@@ -1 +1,467 @@
...
+
+    - router is strongly binded to the apps normally, it is not fit here.
```
Comment (SunShinewyf) [inline docs/source/en/advanced/plugin.md]: binded -> bound (注意时态变换） apps -> application (不建议简写）

File: docs/source/en/advanced/plugin.md [hunk 9]
```diff
@@ -1 +1,467 @@
...
+    - one application might be get a plenty of depending plugins, routers of plugin is very possible conflict with others.it will be a disaster.
```
Comment (SunShinewyf) [inline docs/source/en/advanced/plugin.md]: might be get -> might have 
routers of plugin is -> routers of plugin are (注意单复数）

File: docs/source/en/advanced/plugin.md [hunk 10]
```diff
@@ -1 +1,467 @@
...
+    - if you really need a generally router, you should make it the middleware of the plugins.
```
Comment (SunShinewyf) [inline docs/source/en/advanced/plugin.md]: 后半句感觉有点怪怪的，建议：
you could implement this by adding a middleware in the plugin, 类似这种

File: docs/source/en/advanced/plugin.md [hunk 11]
```diff
@@ -1 +1,467 @@
...
+
+2. the specific information of plugin should be declared in the  `package.json`  by `eggPlugin`：
+
+    - `{String} name` - plugin name(required), it must be unique, it will be used in the configuration of the dependencies of plugin.
+    - `{Array} dependencies` -  the required depending plugins of current plugin(if one of these plugin here missing, the start of this plugin will be failure)
```
Comment (SunShinewyf) [inline docs/source/en/advanced/plugin.md]: 这句感觉和中文不太符合，而且有点问题，末尾的 will be failure ( be 后面只能接 adj !!)


当前插件强依赖的插件列表（如果依赖的插件没找到，应用启动失败）
A list of plugins that are strongly dependent on current plugin (if the dependent plugin is not found, application startup will be failed)

File: docs/source/en/advanced/plugin.md [hunk 12]
```diff
@@ -1 +1,467 @@
...
+    - `{Array} optionalDependencies` - this optional dependencies of this plugin.(if these plugins is not activated, only warnings, the application will rise as normal).
```
Comment (SunShinewyf) [inline docs/source/en/advanced/plugin.md]: if these plugins is -> if these plugins are (注意单复数）
后半句的表述建议改成：
only the warnings will be occurred and not affect the startup of the application

File: docs/source/en/advanced/plugin.md [hunk 13]
```diff
@@ -1 +1,467 @@
...
+    - `{Array} env` - this option is avaliable only when specified the enviorment. the list of env please refer to [env](../basics/env.md). this is optional, most time you can leave it.
```
Comment (SunShinewyf) [inline docs/source/en/advanced/plugin.md]: avaliable -> available
enviormant -> environment (注意单词拼写）
最后的 leave 感觉不太贴合，建议：most time you don't need configure it

File: docs/source/en/advanced/plugin.md [hunk 14]
```diff
@@ -1 +1,467 @@
...
+    ```
+
+  3. No `plugin.js`：
+
+    - `eggPlugin.dependencies` is for declaration of dependencies only, not for importing ,nor activating.
+    - if you want to manage multiple plugins,you should do it in[framework](./framework.md)
+
+## Dependencies Management of Plugins
+
+The dependencies is managed by plugin himself, this is different from middleware.Before loading plugins, the application will read `eggPlugin > dependencies` and `eggPlugin > optionalDependencies` from `package.json` of the plugins, and then sort out the loading orders accordingly, for example, the loading order of the following plugings is `c => b => a`
```
Comment (SunShinewyf) [inline docs/source/en/advanced/plugin.md]: The dependencies is managed -> The dependencies are managed
and then sort out the loading orders accordingly -> and then sort out the loading orders according to the dependency relationship
the loading order of the following plugings is -> the loading order of the following plugins is(单词拼写）

File: docs/source/en/advanced/plugin.md [hunk 15]
```diff
@@ -1 +1,467 @@
...
+  "name": "egg-plugin-c",
+  "eggPlugin": {
+    "name": "c"
+  }
+}
+```
+
+** Attention: the value in the `dependencies` and `optionalDependencies` is the `eggPlugin.name` of plugin, not `package.name`. **
+
+the `dependencies` and `optionalDependencies` is studied from `npm`, most time we are using `dependencies`,it is recommanded.there are about 2 situation to apply the `optionalDependencies`:
```
Comment (SunShinewyf) [inline docs/source/en/advanced/plugin.md]: recommanded -> recommended (单词）
there are about 2 situation -> there are about two situations (阿拉伯数字建议不要出现，注意单复数）

File: docs/source/en/advanced/plugin.md (hunk 11 above)
Comment (AdoBeatTheWorld) [reply docs/source/en/advanced/plugin.md]: A list of plugins that are strongly dependent on current plugin 
总感觉这句的逻辑有点反了：强依赖当前插件的插件列表

Comment (SunShinewyf) [reply docs/source/en/advanced/plugin.md]: 嗯嗯（汗），应该是强调“强依赖”这个动作~

File: docs/source/en/advanced/plugin.md (hunk 13 above)
Comment (AdoBeatTheWorld) [reply docs/source/en/advanced/plugin.md]: 英文要尽量少用否定式的语法，否定式的语气比较重

**Comment by SunShinewyf** [comment]
image

@AdoBeatTheWorld 我用vscode打开你翻译后的文件，随便一看都能看到很多单词错误的地方（绿色的位置差不多都是，除了代码上的），建议用vscode打开，然后装个单词检查的插件~

**Comment by atian25** [comment]
先合了，@SunShinewyf 接手再改。

**Comment by SunShinewyf** [comment]
ok~

**Pull request merged**